lib/devices/Roborock.js

```diff
@@ -115,14 +120,18 @@ class Roborock extends MiioVacuum {
 
 
     updateStatus(res) {
+        if (!this.fanSpeeds) {
+            this.detectFanSpeeds(res["msg_ver"]);
+        }
+
         this.status.state = STATUS_MAP[res["state"]];
         this.status.battery = res["battery"];
         this.status.clean_time = res["clean_time"];
         this.status.clean_area = res["clean_area"];
         this.status.error_code = res["error_code"];
         this.status.map_present = res["map_present"];
         this.status.in_cleaning = res["in_cleaning"];
-        this.status.fan_power = res["fan_power"];
+        this.status.fan_power = Object.keys(this.fanSpeeds).find(key => this.fanSpeeds[key].value === res["fan_power"]);
```

Comment:
I’d argue, `Status.fan_power` should map to an arbitrary percentage value (e.g. 0-100) rather than the device-specific preset?
The UI would gather the presets, but otherwise the API purposes we’d just deal with the percentage value?
While at it: should also document `Status.fan_power`.

Reply:
+1 to that

Reply:
It looks as if the 2008 firmware no longer supports the percent values.
Then I have to map the presets to the old values.

Reply:
The web interface only uses the presets and mqtt also only used the names before.
I think we should stay with names instead of percentage values.

Reply:
I think it would be okay to have a fixed set of valetudo vacuum fan_power constants and just map the device specific ones to them.

Like Min, Silent, Normal, Max and Turbo for example. If the device supports less, we only provide some of those. If it supports more we ignore the additional options.

Reply:
done